python/r2api/r2api.py

```diff
@@ -0,0 +1,159 @@
+from .base import R2Base, Result, ResultArray
+from .debugger import Debugger
+from .file import File
+
+try:
+	import r2pipe
+except ImportError:
+	print 'r2pipe not found'
+	print 'You can install it with pip: pip install r2pipe'
+	raise ImportError('r2pipe not found')
+
+class Function(R2Base):
+	def __init__(self, r2, addr):
+		super(Function, self).__init__(r2)
+
+		self.offset = addr
+
+	def analyze(self):
+		self._exec('af %s' % self.offset)
+
+	def info(self):
+		# XXX: Is this [0] always correct?
+		res = self._exec('afij @ %s' % self.offset, json=True)[0]
+		return Result(res)
+
+	def rename(self, name):
+		self._exec('afn %s %s' % (name, self.offset))
+
+	@property
+	def name(self):
+		return self.info().name
+
+	@name.setter
+	def name(self, value):
+		self.rename(value)
+
+class R2Api(R2Base):
+	def __init__(self, filename=None, r2=None):
+		if filename is not None:
+			r2 = r2pipe.open(filename)
+		super(R2Api, self).__init__(r2)
+
+		self.debugger = Debugger(r2)
+
+		self.info = lambda: Result(self._exec('ij', json=True))
+		self.searchIn = lambda x: self._exec('e search.in=%s'%(x))
+		self.analyzeAll = lambda: self._exec('aaa')
+		self.analyzeCalls = lambda: self._exec('aac')
+		self.basicBlocks = lambda: ResultArray(self._exec('afbj', json=True))
+		self.xrefsAt = lambda: ResultArray(self._exec('axtj %s' % self._tmp_off, json=True))
+		self.refsTo = lambda: ResultArray(self._exec('axfj', json=True))
+		self.opInfo = lambda: ResultArray(self._exec('aoj %s' % self._tmp_off, json=True))[0]
+		self.seek = lambda x: self._exec('s %s'%(x))
+
+	def open(self, filename, at='', perms=''):
+		# See o?
+		self._exec('o %s %s %s' % (filename, at, perms))
+
+	@property
+	def files(self):
+		files = self._exec('oj', json=True)
+		files_lst = []
+		for f in files:
+			files_lst.append(File(self.r2, f['fd']))
+
+		return files_lst
+
+	def functionAt(self, at):
+		res = self._exec('afo %s' % at)
+		if res == '':
+			return None
+		else:
+			return int(res, 16)
+
+	def currentFunction(self):
+		at = '$$'
+		if self._tmp_off != '':
+			at = self._tmp_off.split()[1]
+
+		self._tmp_off = ''
+		return self.functionAt(at)
+
+	def functions(self):
+		res = self._exec('aflj', json=True)
```

Comment:
maybe append `or []` and then use map instead of for+append in 1 line instead of 5